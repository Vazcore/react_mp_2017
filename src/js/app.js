import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import Header from './header'
import Main from './main'

const paddingReset = {paddingLeft: 0, paddingRight: 0};

const App = () => (
  <Grid fluid={true} style={paddingReset}>
    <Row className="show-grid">
      <Col xs={12} md={12}>
        <Header />
      </Col>
      <Col xs={12} md={12}>
        <Main />
      </Col>
    </Row>
  </Grid>
)

export default App