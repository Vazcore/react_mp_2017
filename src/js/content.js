import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import Header from './header'
import Main from './main'
import Footer from './footer'
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { setSearchCriteria, setSortCriteria, changeSearchCriteria, changeSortCriteria } from './actions/criterias'
import {search, sort } from './search/criterias'

class Content extends React.Component {
  constructor(props) {
    super(props)

    this.props.setSearchCriteria(search)
    this.props.setSortCriteria(sort)

    this.props.changeSearchCriteria(search[0])
    this.props.changeSortCriteria(sort[1])
  }
  render() {
    return (
      <Row className="show-grid">
        <Col xs={12} md={12}>
          <Header />
        </Col>
        <Col xs={12} md={12}>
          <Main />
        </Col>
        <Footer />
      </Row>
    )
  }
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({
    setSearchCriteria,
    setSortCriteria,
    changeSearchCriteria,
    changeSortCriteria
  }, dispatch);
}

function mapStateToProps(state) {
  return {
    movies: state.movies
  };
}


export default connect(mapStateToProps, matchDispatchToProps)(Content)