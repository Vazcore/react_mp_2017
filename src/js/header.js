import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';
import SearchBar from './search/searchBar';
import FilmHeader from './film/filmHeader';
import FilmSubHeader from './film/filmSubHeader';
import SubHeaderSearchInfo from './search/searchInfo'

import { HeaderBlock } from '../style/header';
import CommonStyles from '../style/common';
import { Grid, Row, Col } from 'react-bootstrap';

const paddingReset = {paddingLeft: 0, paddingRight: 0};

class Header extends React.Component {
  render() {
    return (
      <header style={HeaderBlock.block}>
     ` <div style={HeaderBlock.wrapper}>
        <Grid style={Object.assign({}, paddingReset, CommonStyles.block)}>
          <Row className="show-grid">
            <Col xs={6} md={6} style={paddingReset}>
              <nav style={HeaderBlock.navigation}>
                <Link style={CommonStyles.nav_link} to='/'>netflixroulette</Link>
              </nav>
            </Col>
            <Switch>
              <Route exact path='/' component={SearchBar} />
              <Route path='/search/:keyword/:criteria' component={SearchBar} />
              <Route path='/film' component={FilmHeader} />
            </Switch>
          </Row>
        </Grid>
        <div style={CommonStyles.subheader}>
          <Grid>
            <Row className="show-grid">
              <Col xs={12} md={12}>
              <Switch>            
                <Route path='/search/:keyword/:criteria' component={SubHeaderSearchInfo} />
                <Route path='/film' component={FilmSubHeader} />
              </Switch>
              </Col>
            </Row>
          </Grid>
        </div>
      </div>
    </header>
    )
  }
}

export default Header
