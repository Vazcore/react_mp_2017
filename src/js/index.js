import React from 'react';
import { render, hydrate } from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom'
import App from './app.js';
import { Provider } from 'react-redux';
import createStore from './reducers/index';

const store = createStore(window.__APP_STORE__);
delete window.__APP_STORE__;

hydrate(
  <Provider store={store}>
  <Router>
    <App />
  </Router></Provider>, document.querySelector('#app'));