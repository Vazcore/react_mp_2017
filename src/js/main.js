import React from 'react'
import { Switch, Route } from 'react-router-dom';
import Home from './home/home';
import Search from './search/search'
import Film from './film/film';
import { withRouter } from 'react-router-dom'
import NotFound from './notFound';
import { Grid, Row, Col } from 'react-bootstrap';

import commonStyles from '../style/common'

class Main extends React.PureComponent {
  constructor(props) {
    super(props)
  }

  render() {
    return(
    <Grid>
      <Row className="show-grid">
        <Col xs={12} md={12}>
          <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/search/:keyword/:criteria' component={Search}/>
            <Route path='/film' component={Film}/>
            <Route path='*' component={NotFound}/>
          </Switch>
        </Col>
      </Row>
    </Grid>
    )
  }
}


export default withRouter(Main)