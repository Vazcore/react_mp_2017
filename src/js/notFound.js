import React from 'react'
import commonStyles from '../style/common'

class Footer extends React.PureComponent {
  render() {
    return (
      <div style={commonStyles.formGroup}>
        <div style={commonStyles.panelDescription}>Not Found</div>
      </div>
    )
  } 
}

export default Footer;