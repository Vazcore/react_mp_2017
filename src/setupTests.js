import { configure } from 'enzyme';
import 'raf-polyfill';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });