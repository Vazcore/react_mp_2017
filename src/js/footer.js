import React from 'react'
import { Switch, Route, Link } from 'react-router-dom'
import { HeaderBlock } from '../style/header'
import commonStyles from '../style/common'

class Footer extends React.PureComponent {
  render() {
    return (
      <div style={commonStyles.footer}>
        <nav style={HeaderBlock.navigation}>
          <Link style={commonStyles.nav_link} to='/'>netflixroulette</Link>
        </nav>
      </div>
    )
  } 
}

export default Footer;