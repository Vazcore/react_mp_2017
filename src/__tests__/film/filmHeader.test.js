import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import FilmHeader from '../../js/film/filmHeader';
import '../../setupTests.js';

const movie = {
  id: 1,
  poster_path: 'path1',
  title: 'Title1',
  release_date: new Date(2017, 1, 1),
  category: 'Horor',
  credits: {
    crew: []
  }
};


jest.mock('react-redux', () => {
  return {
    connect: () => {
      return componentClass => componentClass
    }
  };
});
jest.mock('react-router-dom', () => {
  return {
    withRouter: componentClass => componentClass,
    Link: (props) => <span>{props.children}</span>
  }
});

jest.mock('react-bootstrap', () => {
  return {
    Row: (props) => <span>{props.children}</span>,
    Col: (props) => <span>{props.children}</span>
  };
});
jest.mock('../../js/helpers/api', () => {
  const movie = {
    id: 1,
    poster_path: 'path1',
    title: 'Title1',
    release_date: new Date(2017, 1, 1),
    category: 'Horor',
    credits: {
      crew: []
    }
  };
  return {
    getMoviesByPerson: jest.fn(),
    getMovieDetails: () => new Promise((resolve, reject) => resolve(movie)),
    getCredits: () => new Promise((resolve, reject) => resolve({}))
  };
});
jest.mock('../../js/helpers/dates', () => {
  return {
    dateStringToYear: jest.fn(),
    findDirector: () => ({})
  };
});
jest.mock('../../js/helpers/img', () => {
  return {
    getImg: (path) => <span>{path}</span>
  };
})

const props = {
  match: {
    params: { title: 'Title1' }
  },
  selectMovie: jest.fn(),
  selectedMovie: movie
};

describe('FilmHeader component', () => {  
  test('FilmHeader should render', () => {
    const component = renderer.create(
      <FilmHeader {...props}  />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });  
});

describe('FilmHeader mount', () => {
  // const component = mount(
  //   <FilmHeader {...props}  />
  // );
  // const instance = component.instance();
  // instance.foundMovie('Title2');
  // expect(instance.movie).toBe(movie.id);
});