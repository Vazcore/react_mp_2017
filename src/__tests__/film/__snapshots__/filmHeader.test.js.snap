// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmHeader component FilmHeader should render 1`] = `
<div>
  <span>
    Search
  </span>
  <span>
    <span>
      <span>
        <span>
          path1
        </span>
      </span>
      <span>
        <p
          style={
            Object {
              "color": "#F55263",
              "fontSize": "34px",
            }
          }
        >
          Title1
          <span
            style={
              Object {
                "border": "1px solid #99C16F",
                "borderRadius": "50px",
                "color": "#99C16F",
                "display": "inline-block",
                "fontSize": "20px",
                "height": "50px",
                "padding": "10px",
                "textAlign": "center",
                "verticalAlign": "middle",
                "width": "50px",
              }
            }
          />
        </p>
        <p
          style={
            Object {
              "color": "#C8C8C8",
              "fontSize": "19px",
            }
          }
        />
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "19px",
              "fontWeight": "bold",
            }
          }
        >
          <span
            style={
              Object {
                "display": "inline-block",
                "marginRight": "30px",
              }
            }
          />
        </p>
        <p
          style={
            Object {
              "color": "#C8C8C8",
              "fontSize": "19px",
              "marginTop": "40px",
            }
          }
        />
      </span>
    </span>
  </span>
</div>
`;
